packages/components/src/tools-panel/tools-panel/hook.js

```diff
@@ -93,10 +119,11 @@ export function useToolsPanel( props ) {
 		}
 
 		// Turn off display of all non-default items.
-		const resetMenuItems = {};
+		const resetMenuItems = { default: {}, optional: {} };
 
 		panelItems.forEach( ( { label, isShownByDefault } ) => {
-			resetMenuItems[ label ] = !! isShownByDefault;
+			const group = isShownByDefault ? 'default' : 'optional';
+			resetMenuItems[ group ][ label ] = false;
```

Comment:
Small nit: since this same logic is repeated earlier in this same file, I wonder if it could be extracted to improve the code's resistance to future changes (even by just creating a function private to this file)